### PR TITLE
[WIP] [RouteResolver] Should be less opinionated

### PR DESCRIPTION
**Pull request by renatomefi**

### CODE REVIEW

File: Config/GraphQLEndpoint/GraphQLEndpointInvalidSchemaException.php [hunk 1]
```diff
@@ -2,10 +2,15 @@
...
+class GraphQLEndpointInvalidSchemaException extends \LogicException
 {
     public static function forSchemaAndResolver($schemaName, $resolverClass)
     {
         return new static('Schema "'.$schemaName.'" isn\'t valid for resolver "'.$resolverClass.'"');
     }
+
+    public static function forNoSchemaAndResolver($resolverClass)
+    {
+        return new static('NO schema provided for resolver "'.$resolverClass.'"');
```
Comment (ooflorent) [inline Config/GraphQLEndpoint/GraphQLEndpointInvalidSchemaException.php]:
```diff
- NO
+ No

File: Config/GraphQLEndpoint/Helpers/OverblogGraphQLBundleEndpointResolver.php [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 final class OverblogGraphQLBundleEndpointResolver
 {
-    public static function getByName($name)
+    public static function getByName($name = null)
     {
-        if ('default' === $name) {
+        if ('default' === $name || null === $name) {
```
Comment (ooflorent) [inline Config/GraphQLEndpoint/Helpers/OverblogGraphQLBundleEndpointResolver.php]: I think we should use only `'default'` or `null`. Not both.
Depending on the decision, the default value of `$name` should be changed.

File: Config/GraphQLEndpoint/RootResolver.php [hunk 3]
```diff
@@ -6,9 +6,9 @@
 
 final class RootResolver implements GraphiQLControllerEndpoint
 {
-    public function getBySchema($name)
+    public function getBySchema($name = null)
     {
-        if ('default' === $name) {
+        if ('default' === $name || null === $name) {
```
Comment (ooflorent) [inline Config/GraphQLEndpoint/RootResolver.php]: Same here

File: Config/GraphQLEndpoint/RootResolver.php [hunk 4]
```diff
@@ -17,6 +17,6 @@ public function getBySchema($name)
 
     public function getDefault()
     {
-        return $this->getBySchema('default');
+        return $this->getBySchema();
```
Comment (ooflorent) [inline Config/GraphQLEndpoint/RootResolver.php]: I'm 👎  on this one. Explicit behavior is better.

File: Config/GraphQLEndpoint/RouteResolver.php [hunk 5]
```diff
@@ -23,27 +23,43 @@ public function __construct(RouterInterface $router, array $routeCollection)
...
+        $route = array_key_exists($name, $this->routeCollection) ? $this->routeCollection[$name] : null;
 
-        return $this->router->generate(...$route);
+        return $this->generateRoute($route, $name);
     }
 
     public function getDefault()
     {
-        return $this->getBySchema('default');
+        return $this->getBySchema();
```
Comment (ooflorent) [inline Config/GraphQLEndpoint/RouteResolver.php]: I don't like this

File: Tests/Config/GraphQLEndpoint/RouteResolverTest.php [hunk 6]
```diff
@@ -37,7 +37,7 @@ public function testInvalidRoute()
         $routeResolver = $this->subject([]);
 
         $this->expectException(GraphQLEndpointInvalidSchemaException::class);
-        $this->expectExceptionMessage('Schema "default" isn\'t valid for resolver "Overblog\GraphiQLBundle\Config\GraphQLEndpoint\RouteResolver"');
+        $this->expectExceptionMessage('NO schema provided for resolver "Overblog\GraphiQLBundle\Config\GraphQLEndpoint\RouteResolver"');
```
Comment (ooflorent) [inline Tests/Config/GraphQLEndpoint/RouteResolverTest.php]:
```diff
- NO
+ No

**Comment by renatomefi** [comment]
@ooflorent I understand your feeling about the 'default' part, the idea is the RouteResolver to be not influenced by how OverblogGraphQLBundle works, and 'default' is sort of this, so the idea is to make the RouteResolver usable for anything, even the user configuring manual routes inside of it, that's the objective.
Having it in mind do you have a suggestion on how to achieve it?

**Comment by ooflorent** [comment]
I'm ok with `default`. And I'm also ok with `null`. I'm just saying we should only one option. Since both behave the same, I don't really care of the chosen one.

**Pull request closed without merging**